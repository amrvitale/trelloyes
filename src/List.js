//list component should render markup matching design.html
//section wtih class of 'List' contain header and div with class of 'List-cards'
//List component accepts 2 props: header and cards
//header = string of header of card to render
// cards is array of card objects, each object has title and content
//props will be passed in for each list from app component
//List should render card componentfor each of cards in cards array prop
//Each instance of card componnent should be passed 2 props and a key
//2 props = title and content
//title prop is string for card's title
// content prop is string of card's content
import React from 'react';
import Card from './Card'
import './List.css';

export default function List(props) {
  return (
    <section className='List'>
      <header className='List-header'>
        <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {props.cards.map((card) =>
          <Card
            key={card.id}
            id={card.id}
            title={card.title}
            content={card.content}
            onClickDelete={props.onClickDelete}
          />
        )}
        <button
          type='button'
          className='List-add-button'
          onClick={() => props.onClickAdd(props.id)}
        >
          + Add Random Card
        </button>
      </div>
    </section>
  )
}

List.defaultProps = {
  onClickAdd: () => {},
}
