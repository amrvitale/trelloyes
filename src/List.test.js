import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import List from './List';

describe ('List component', () => {
  it('renders without crashing', () => {
    const section = document.createElement('section');
    ReactDOM.render(
      <List
      key ='0'
      header="Test Header"
      cards={["1", "2"]}
      />, section);

    ReactDOM.unmountComponentAtNode(section);
  });

  it ('renders the UI as expected', () => {
  const tree = renderer
    .create(<List key = '0' header = 'test' cards ={["1"]}/>)
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
