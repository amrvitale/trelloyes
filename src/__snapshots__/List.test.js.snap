// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3 />
      <p />
      <button
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
</section>
`;
